stage0/src/asm/boot.s

```diff
@@ -167,33 +142,24 @@ _protected_mode_start:
     xor %ecx, %ecx            # ECX = 0 - we're not interested in a subpage
     cpuid                     # EAX, EBX, ECX, EDX = CPUID(EAX, ECX)
     and $0b111111, %ebx       # zero out all but EBX[5:0], which the C-bit location
-    push %ebx                 # save the full C-bit location value for later
+    mov %ebx, %esi            # save the full C-bit location value for later to pass into main
```

Comment:
We no longer need  EDI for any particular purpose (like `rep stosb`), so instead of putting the C-bit location in `ESI` here and moving it into `EDI` later, we could just put it in `EDI` directly here and use `ESI` for the bitmask below.

Reply:
Good point. I will change that.